tests/Unit/Domain/CoordinateMaps/Test_UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,517 @@
+        // edge of sphere 2.  But don't let max_rho_sphere3 be negative.
+        const double max_rho_sphere3 =
+            std::max(0.0, radius_two * 0.95 - radius_one);
+
+        // Alpha always gets smaller when rho gets larger (for other
+        // quantities fixed). So if alpha < min_alpha even when rho=0, then
+        // there is no hope.  We always fail.
+        const double alpha_plus_if_rho_is_zero =
+            atan2(z_plane_plus_two - z_plane_plus_one,
+                  radius_one * sin(theta_max_plus_one) -
+                      radius_two * sin(theta_max_plus_two));
+        CHECK(alpha_plus_if_rho_is_zero >= min_alpha_plus);
+        const double alpha_minus_if_rho_is_zero =
+            atan2(z_plane_minus_one - z_plane_minus_two,
+                  radius_one * sin(theta_max_minus_one) -
+                      radius_two * sin(theta_max_minus_two));
+        CHECK(alpha_minus_if_rho_is_zero >= min_alpha_minus);
+
+        const double max_rho_alpha_plus_first_term =
+            abs(min_alpha_plus - 0.5 * M_PI) < 1.e-4
+                ? 0.0
+                : (z_plane_plus_two - z_plane_plus_one) / tan(min_alpha_plus);
+
+        const double max_rho_alpha_minus_first_term =
+            abs(min_alpha_minus - 0.5 * M_PI) < 1.e-4
+                ? 0.0
+                : (z_plane_minus_one - z_plane_minus_two) /
+                      tan(min_alpha_minus);
+
+        // maximum rho so that the alpha condition is satisfied
+        const double max_rho_alpha_plus = max_rho_alpha_plus_first_term -
+                                          radius_one * sin(theta_max_plus_one) +
+                                          radius_two * sin(theta_max_plus_two);
+        const double max_rho_alpha_minus =
+            max_rho_alpha_minus_first_term -
+            radius_one * sin(theta_max_minus_one) +
+            radius_two * sin(theta_max_minus_two);
+        const double max_rho =
+            std::min({max_rho_sphere, max_rho_sphere2, max_rho_sphere3,
+                      max_rho_alpha_plus, max_rho_alpha_minus});
+        CHECK(max_rho >= 0.0);
+        return unit_dis(gen) * max_rho;
+      }();
+
+  const double phi = angle_dis(gen);
+  const std::array<double, 3> center_one = {
+      center_two[0] + horizontal_distance_spheres * cos(phi),
+      center_two[1] + horizontal_distance_spheres * sin(phi), center_one_z};
+  CAPTURE(center_one);
+
+  const CoordinateMaps::UniformCylindricalSide map(
+      center_one, center_two, radius_one, radius_two, z_plane_plus_one,
+      z_plane_minus_one, z_plane_plus_two, z_plane_minus_two);
+  test_suite_for_map_on_cylinder(map, 1.0, 2.0, true, true);
+
+  // The following are tests that the inverse function correctly
+  // returns an invalid std::optional when called for a point that is
+  // outside the range of the map.
+
+  // Point with z > z_plane_plus_two.
+  CHECK_FALSE(map.inverse({{0.0, 0.0, z_plane_plus_two + 1.0}}));
```

Comment:
Explicitly call `.has_value()` here rather than implicitely converting an optional to a bool. Same for the rest fo these checks also